zerver/actions/users.py

```diff
@@ -517,24 +517,76 @@ def do_change_user_role(
     send_stream_events_for_role_update(user_profile, previously_accessible_streams)
 
 
+@transaction.atomic(savepoint=False)
```

Comment:
I think we determined `durable=True` is the preferred form to use by default; we only want `savepoint=False` for specific use cases, because of how unit tests work.

Reply:
I'm not sure it's this straight-forward for these functions - `durable=True` makes sense when we want to protect the function from being called inside another atomic block. Not sure we want that protection here. And for `do_change_can_forge_sender` - we actually call it inside another atomic block in `do_create_internal_realm`: https://github.com/zulip/zulip/blob/d8dd682944838c71468e8498889e840d6b384515/zerver/lib/server_initialization.py#L77

So adding `durable=True` causes that to fail. The other functions aren't called like that, but do we want to have inconsistencies between them? `savepoint=False` just makes sure Django doesn't create savepoints when not needed, avoiding that bit of overhead; so it's quite harmless and not opinionated.

Reply:
The issue with the other is that it interacts oddly with transaction rollback in tests, but maybe it's fine; the fact that you didn't need test changes for this PR suggests it's not important here.